docs/current/guides/snippets/azure-pipelines-container-instances/main.py

```diff
@@ -0,0 +1,97 @@
+import sys
+import os
+
+import anyio
+import dagger
+from azure.identity import DefaultAzureCredential
+from azure.mgmt.containerinstance import ContainerInstanceManagementClient
+
+# configure container group, name and location
+container_name = "my-app";
+container_group_name = "my-app";
+container_group_location = "eastus";
+resource_group_name = "my-group";
+
+async def main():
+
+    # check for required variables in host environment
+    for var in ["DOCKERHUB_USERNAME", "DOCKERHUB_PASSWORD", "AZURE_SUBSCRIPTION_ID", "AZURE_TENANT_ID", "AZURE_CLIENT_ID", "AZURE_CLIENT_SECRET"]:
+        if var not in os.environ:
+            raise EnvironmentError('"%s" environment variable must be set' % var)
+
+    # initialize Dagger client
+    async with dagger.Connection(dagger.Config(log_output=sys.stderr)) as dagger_client:
+
+        # set registry password as Dagger secret
+        secret = dagger_client.set_secret("password", os.environ.get("DOCKERHUB_PASSWORD"));
+
+        # get reference to the project directory
+        source = (
+            dagger_client
+            .host()
+            .directory(".", exclude=["node_modules", "ci"])
+        )
+
+        # get Node image
+        node = (
+            dagger_client
+            .container(platform=dagger.Platform("linux/amd64"))
+            .from_("node:18")
+        )
+
+        # mount source code directory into Node image
+        # install dependencies
+        # set entrypoint
+        c = (
+            node
+            .with_directory("/src", source)
+            .with_workdir("/src")
+            .with_exec(["cp", "-R", ".", "/home/node"])
```

Comment:
Can avoid the first `with_workdir`:

```suggestion
            .with_exec(["cp", "-R", "./src", "/home/node"])
```

Reply:
This didn't seem to work when I tested it. I ran out of time to investigate why. I'll leave it as is for the moment.

Reply:
This implied removing `.with_workdir("/src")`